pkg/driverbuilder/builder/sle.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2023 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	_ "embed"
+
+	"github.com/falcosecurity/driverkit/pkg/kernelrelease"
+)
+
+//go:embed templates/sle.sh
+var sleTemplate string
+
+// TargetTypeSLE identifies the sle target.
+const TargetTypeSLE Type = "sle"
+
+// sle is a driverkit target.
+type sle struct {
+}
+
+func init() {
+	byTarget[TargetTypeSLE] = &sle{}
+}
+
+type sleTemplateData struct {
+	commonTemplateData
+	KernelPackage string
+}
+
+func (v *sle) Name() string {
+	return TargetTypeSLE.String()
+}
+
+func (v *sle) TemplateScript() string {
+	return sleTemplate
+}
+
+func (v *sle) URLs(kr kernelrelease.KernelRelease) ([]string, error) {
```

Comment:
Is there no way to fetch headers?

Reply:
Not currently - SLE is an enterprise distro that requires authentication to get headers. For now, I followed what we did for RHEL, which is allow the user to bring-their-own-authenticated-container to build with `--builderimage`.

I'm working on an MR today for kernel-crawler to do the same as RHEL there too - allow the user to bring an auth'd container.

Reply:
Oh i see, thanks! :D